src/main.cpp

```diff
@@ -1,30 +1,143 @@
 #include "../lib/core/types.hpp"
+#include "../lib/logical_solver/presolve.hpp"
```

Comment:
Ok I know this is an annoying one but for now, make a separate file for now called something like `run_logical_solver` or similar and wrap all this in a function or we'll keep getting big problems in the main.cpp file.

Reply:
Done.